File: mobile/src/main/kotlin/de/scout/fireplace/settings/NumberRangeInputFilter.java
package de.scout.fireplace.settings;

import android.text.InputFilter;
import android.text.Spanned;
import android.util.Log;

public class NumberRangeInputFilter implements InputFilter {

  private final int minimum;
  private final int maximum;

  NumberRangeInputFilter(int minimum, int maximum) {
    this.minimum = minimum;
    this.maximum = maximum;
  }

  @Override
  public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
    try {
      int input = Integer.parseInt(dest.toString() + source.toString());
      if (isInRange(minimum, maximum, input)) {
        return null;
      }
    } catch (NumberFormatException exception) {
      Log.e("NUMBER_FORMAT", exception.getMessage(), exception);
    }

    return "";
  }

  private boolean isInRange(int a, int b, int c) {
    return b > a ? c >= a && c <= b : c >= b && c <= a;
  }
}

File: feature/src/main/kotlin/de/scout/fireplace/feature/settings/NumberRangeInputFilter.kt
package de.scout.fireplace.feature.settings

import android.text.InputFilter
import android.text.Spanned

internal class NumberRangeInputFilter internal constructor(private val minimum: Int, private val maximum: Int) : InputFilter {

  override fun filter(source: CharSequence, start: Int, end: Int, dest: Spanned, dstart: Int, dend: Int): CharSequence? {
    try {
      val input = Integer.parseInt(dest.toString() + source.toString())
      if (isInRange(minimum, maximum, input)) {
        return null
      }
    } catch (ignored: NumberFormatException) {
    }
    return ""
  }

  private fun isInRange(a: Int, b: Int, c: Int): Boolean {
    return if (b > a) c in a..b else c in b..a
  }
}
